ui/src/components/UploadStory.tsx

```diff
@@ -0,0 +1,855 @@
+    const theme = useTheme();
+    const smUp = useMediaQuery(theme.breakpoints.up("sm"), { noSsr: true });
+    const itemCount = itemData.length;
+    const itemSize = smUp ? 36 : 48;
+
+    const getChildSize = (child: React.ReactNode) => {
+      if (React.isValidElement(child) && child.type === ListSubheader) {
+        return 48;
+      }
+
+      return itemSize;
+    };
+
+    const getHeight = () => {
+      if (itemCount > 8) {
+        return 8 * itemSize;
+      }
+      return itemData.map(getChildSize).reduce((a, b) => a + b, 0);
+    };
+
+    const gridRef = useResetCache(itemCount);
+
+    return (
+      <div ref={ref}>
+        <OuterElementContext.Provider value={other}>
+          <VariableSizeList
+            itemData={itemData}
+            height={getHeight() + 2 * LISTBOX_PADDING}
+            width="100%"
+            ref={gridRef}
+            outerElementType={OuterElementType}
+            innerElementType="ul"
+            itemSize={(index) => getChildSize(itemData[index])}
+            overscanCount={5}
+            itemCount={itemCount}
+          >
+            {renderRow}
+          </VariableSizeList>
+        </OuterElementContext.Provider>
+      </div>
+    );
+  }
+);
+
+const renderGroup = (params: AutocompleteRenderGroupParams) => [
+  <ListSubheader key={params.key} component="div">
+    {params.group}
+  </ListSubheader>,
+  params.children,
+];
+
+interface InputProps {
+  onKeyDown: (
+    event: KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => void;
+}
+
+interface TagParameters {
+  inputProps: InputProps;
+}
+
+interface StoryProps {
+  id: number;
+  currentStory: Story;
+  bio: string;
+}
+
+export const UploadStory: React.FC<StoryProps> = ({
+  id,
+  currentStory,
+  bio,
+}: StoryProps) => {
+  const { data: tagOptions, error } = useSWR<string[]>("/api/tags");
+  const [tagArray, setTagArrayValues] = useState(currentStory.tags);
+  const [authorCountry, setAuthorCountry] = useState(
+    currentStory.author_country
+  );
+  const [currentCity, setCurrentCity] = useState(
+    currentStory.current_city.toUpperCase()
+  );
+  const [autocompleteAuthorCountry, setAutocompleteAuthorCountry] = useState(
+    currentStory.author_country
+  );
+  const [addedCountry, setAddedCountry] = useState("");
+  const [newImage, setNewImage] = useState(currentStory.image_url);
+  const [disabled, setDisabled] = useState(false);
+  //handleSubmit component states
+  const [dialogOpenState, setDialogOpenState] = useState(false);
+  const [uploadErrorState, setErrorOpen] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+
+  const { data: countries, error: errorCountries } = useSWR<string[]>(
+    "/api/countries"
+  );
+
+  const yearArray = [];
+  let year: number = new Date().getFullYear();
+  for (let i = 30; i > 0; i--) {
+    yearArray.push(year);
+    year -= 1;
+  }
```

Comment:
A little JS magic :)

```suggestion
const startYear = new Date().getFullYear();
const yearArray = Array.from({ length: 30 }, (_, i) => startYear - i)
```

Reply:
Will do. What exactly is going on here (is there a name for it)? I can tell what Array.from does but not how the interior works.

Reply:
`Array.from` creates an array from an "array-like" object. In this case the object defines a length, so it can be considered array like.

Reply:
Ah cool, thanks!